adr-notification-system.md

```diff
@@ -0,0 +1,33 @@
+# ADR 2: Extract Notification Component
+
+## Status
+
+PROPOSED
+
+## Context
+
+Notification mechanisms are used to notify users and experts about changes in the state of a ticket workflow and to send surveys. Currently, it's implemented as a component in the monolith system and uses external sms and email gateways.
+
+## Decision
+
+1. Move notification component into a separate module.
+
+2. The module should use no underlying data source.
+
+3. The main concern is, provided with all the necessary data to generate notifications, to handle the communication with external gateways, and handle delays, timeouts and exceptions gracefully.
+
+4. Internal modules should use the notification module in a "fire-and-forget" fashion.
+
+## Consequences
+
+1. We can extract the system without any major change and without data migration, therefore, the implementation risks are minimal.
+
+2. It's easy to scale the system up and down on demand.
```

Comment:
I think it's easy to interpret as a generic comment about the system and not just the notification system.
```suggestion
2. It's easy to scale the notification system up and down on demand.
```